coding_style/README.adoc

```diff
@@ -0,0 +1,217 @@
+====
++
+* Do not use the Ansible-specific `yes` and `no` as boolean values in YAML as these are completely custom extensions used by Ansible and are not part of the YAML spec.
+* Avoid comments in playbooks when possible.
+  The task `name` value should be descriptive enough to tell what a task does.
+  Variables should be well commented in the `defaults` and `vars` directories and should, therefore, not need explanation in the playbooks themselves.
+* Use a single space separating the template markers from the variable name inside all Jinja2 template points.
+  For instance, always write it as `{{ variable_name_here }}`.
+  The same goes if the value is an expression. `{{ variable_name | default('hiya, doc') }}`
+* When naming files, use the `.yml` extension and _not_ `.yaml`.
+  `.yml` is what `ansible-galaxy init` does when creating a new role template.
+* Use double quotes for YAML strings with the exception of Jinja2 strings which will use single quotes.
+* Do not use quotes unless you have to, especially for short module-keyword-like strings like `present`, `absent`, etc.
+  But do use quotes for user-side strings such as descriptions, names, and messages.
+
+== Python Guidelines
+
+* Review Ansible guidelines for https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html[modules] and https://docs.ansible.com/ansible/latest/dev_guide/index.html[development].
+* Use https://pep8.org/[PEP8].
+* File headers and functions should have comments for their intent.
+
+== Ansible Guidelines
+
+* Ensure that all tasks are idempotent.
+* https://github.com/ansible/ansible/issues/10374[Ansible variables use lazy evaluation.]
+* Prefix all tags within a role with the role name or, alternatively, a "unique enough" but descriptive prefix.
+* Prefer the command module over the shell module unless you explicitly need shell functionality such as, e.g., piping.
+  Even better, use a dedicated module, if it exists.
+  If not, see the <<check-mode-and-idempotency-issues,section>> about idempotency and check mode and make sure that your task is idempotent and supports check mode properly;
+  your task will likely need options such as `changed_when:` and maybe `check_mode:`).
+* Anytime `command` or `shell` modules are used, add a comment in the code with justification to help with future maintenance.
+* Use the `| bool` filter when using bare variables (expressions consisting of just one variable reference without any operator) in `when`.
+* Do not use `meta: end_play`.
++
+[%collapsible]
+====
+Explanation:: It aborts the whole play instead of a given host (with multiple hosts in the inventory).
+  If absolutely necessary, consider using `meta: end_host`.
+====
++
+* Task names can be made dynamic by using variables (wrapped in Jinja2 templates), this helps with reading the logs.
+* Do not use variables (wrapped in Jinja2 templates) for play names, variables don't get expanded properly there.
```

Comment:
Also not in loops

Reply:
I don't understand this comment

Reply:
Loop variables (by default "item") also don't get properly expanded in looped tasks' name and hence are not to be used there.